### ISSUE TITLE
"The following items are duplicates:" dialog is flawed/useless

### ISSUE THREAD

**Issue by rwperrott** [axcore/tartube#561]
### What operating system are you using?
Linux Mint 21.1 MATE 64-bit

### What version of Tartube are you using?
python3-tartube_2.4.386.deb

###If your bug report is
- It lacks a top window bar, so can't be closed except by Right-click and selecting `Close`, which confused me initially.
- It lacks a horizontal scroll bar, so I can't see all the items, which extend below the bottom of my display.
- The list of items is too terse to see what most of the items are, and including which video they are for.

**Comment by rwperrott** [comment]
I did a database repair, because the target folder contents did not make sense, and it fixed errors. Maybe a database check and repair should be prompted for, but not forced, and indicated, when issues are spotted at start up, however that is a different matter to this bug.

**Comment by rwperrott** [comment]
Next run saw this in shell:
```shell
$ tartube
/usr/lib/python3/dist-packages/gi/overrides/Gio.py:42: Warning: unable to set property 'strikethrough-set' of type 'gboolean' from value of type 'gchararray'
  return Gio.Application.run(self, *args, **kwargs)
/usr/lib/python3/dist-packages/gi/overrides/Gio.py:42: Warning: unable to set property 'underline-set' of type 'gboolean' from value of type 'gchararray'
  return Gio.Application.run(self, *args, **kwargs)
/usr/lib/python3/dist-packages/gi/overrides/Gio.py:42: Warning: unable to set property 'weight-set' of type 'gboolean' from value of type 'gchararray'
  return Gio.Application.run(self, *args, **kwargs)
/usr/lib/python3/dist-packages/gi/overrides/Gio.py:42: Warning: unable to set property 'style-set' of type 'gboolean' from value of type 'gchararray'
  return Gio.Application.run(self, *args, **kwargs)

(tartube:2475598): Gdk-CRITICAL **: 16:25:18.494: gdk_window_thaw_toplevel_updates: assertion 'window->update_and_descendants_freeze_count > 0' failed
^C%                                                                                                                                                               

```

**Comment by rwperrott** [comment]
I'm not sure if this is safe to use as-is, because I may cause further damage to my stored content.

**Comment by axcore** [comment]
The first four lines are Gtk bugs; the rest is output generated by VLC. I don't see anything there generated by Tartube.

**Comment by rwperrott** [comment]
It looks like I'll need to be more wary of GUI python applications, I've already some more in calibre recently; maybe newer python/GTK releases are breaking stuff.

**Comment by axcore** [comment]
>dialog is flawed/useless

Presumably you mean the 'Add videos' dialogue window, which I have now tested on Linux Mint 21.1 MATE 64-bit. Apologies, but I can't see the problems you describe: both the window bar and horizontal/vertical scroll bars are present:

![scrollbars](https://github.com/axcore/tartube/assets/37951926/792aac97-3fbd-4f95-8fcc-884e48c130e6)

I can't explain why you are seeing something different.

>The list of items is too terse to see what most of the items are, and including which video they are for

It's true that the window lacks polish, but I'm not sure how it can be improved. At the moment, it essentially conveys the message "there is 1 duplicate" or "there are many duplicates"; what would be the use case for something more exotic, like a list of thumbnails? Why would a user need to know which particular videos are already in the database?

>I did a database repair, because the target folder contents did not make sense, and it fixed errors. Maybe a database check and repair should be prompted for, but not forced, and indicated, when issues are spotted at start up

Database checks may take a long time, if the database is large. When I'm made aware of an issue that causes problems in the database, the new release of Tartube will automatically perform a database check and repair, probably without the user being aware of it. Problems like that are very tricky to track down, though.

>I'm still seeing duplicates dialog after database repair, which is concerning.

Can you be sure that the obvious explanation is not the correct one, namely that you've already added the same URL to Tartube's database?

>It looks like I'll need to be more wary of GUI python applications, I've already some more in calibre recently; maybe newer python/GTK releases are breaking stuff.

All Gtk releases are a buggy mess; it's one of the hazards of using it. I have spent some considerable time checking that the Gtk warnings generated by Tartube have no effect on functionality; there are a lot of warnings in your output, but they are all familiar ones that don't seem to break anything.

**Comment by rwperrott** [comment]
No, not the "Add Videos" window, but rather when I tried to drag & drop videos in "Missing Videos" folder to a folder I had previously created. If the "Missing Video" folder is a virtual view, which may include items from real folders, then it should have some safeguards to stop overwrite by self, and possible consequential file corruption/deletion {1} {2}.

{1} Calibre has a possibly relevant, nasty bug, where, when I request a book rename on Linux, merely changing the case of one or more letters, thus the filename, on an SMB share of a case-insensitive ZFS dataset directory, the file can end up deleted!
{2} If such damage happened after a ZFS snapshot, I still have, I can undo such damage; I can probably do this for this Tartube video move glitch.

**Comment by axcore** [comment]
I found the cause of the database errors. The "Missing Videos" folder keeps track of the total number of missing videos, but this number is updated incorrectly. 

I have fixed the problem, but it's a very minor issue: nothing depends on the incorrect count, so the database isn't damaged. Doing a database repair will automatically re-calculate the total anyway.

For the other thing: when I drag a video from the **Missing Videos** folder to another folder, **twice**, I see an unexpected duplicate dialogue. Not sure what the cause is yet but I'll investigate further.

![twice](https://github.com/axcore/tartube/assets/37951926/fbd003e9-4197-40bf-adf7-05cf2eb11920)

**Comment by axcore** [comment]
Two pieces of code respond to the drag and drop, when only one should. The "wrong" one is the one that's creating the unexpected duplicated videos. I haven't worked out how to disable it yet.

I have confirmed that Tartube is trying to move the video from its actual location, not from the "virtual" Missing Videos folder, as you suggested it might.

**Comment by axcore** [comment]
The drag and drop issue is fixed in v2.4.402. The new code is uploaded to github/pypi. If you're a Windows user, see [here](https://github.com/axcore/tartube#update-tartube-on-ms-windows) for update instructions.
